### PR TITLE
initial VectorNav (VN-100, VN-200, VN-300) INS support

### PR DESCRIPTION
**Pull request by dagar**
Starting with simple sensor data, but this will quickly evolve to support full external INS integration.

TODO
 - [ ] verify full raw sensor support
 - [ ] configurable rotation
 - [ ] configurable positions (antennas, etc)
 - [ ] VN parameter support
 - [ ] integrate with onboard estimator selection

### CODE REVIEW

**Comment by ryanjAA** [comment]
Great! Is this in a state to test? If so I’ll get one purchased.

File: src/drivers/drv_sensor.h [hunk 1]
```diff
@@ -221,6 +221,10 @@
 #define DRV_POWER_DEVTYPE_INA220 0xD3
 #define DRV_POWER_DEVTYPE_INA238 0xD4
 
+#define DRV_INS_DEVTYPE_VN100 0xD1
```
Comment (dagar) [inline src/drivers/drv_sensor.h:224]:
```suggestion
#define DRV_INS_DEVTYPE_VN100 0xE1
```

File: src/drivers/ins/vectornav/VectorNav.cpp [hunk 2]
```diff
@@ -0,0 +1,657 @@
...
+				      (ImuGroup)(IMUGROUP_ACCEL | IMUGROUP_ANGULARRATE),
+				      GPSGROUP_NONE,
+				      ATTITUDEGROUP_NONE,
+				      INSGROUP_NONE,
+				      GPSGROUP_NONE)
+	   ) {
+		// TIMEGROUP_TIMESTARTUP
+		uint64_t time_startup = VnUartPacket_extractUint64(packet);
+
+		// IMUGROUP_ACCEL
```
Comment (dagar) [inline src/drivers/ins/vectornav/VectorNav.cpp:119]:
```suggestion
		// IMUGROUP_ACCEL -> IMUGROUP_UNCOMPACCEL
```

File: src/drivers/ins/vectornav/VectorNav.cpp [hunk 3]
```diff
@@ -0,0 +1,657 @@
...
+		vec3f accel = VnUartPacket_extractVec3f(packet);
+
+		// IMUGROUP_ANGULARRATE
```
Comment (dagar) [inline src/drivers/ins/vectornav/VectorNav.cpp:122]:
```suggestion
		// IMUGROUP_ANGULARRATE -> IMUGROUP_UNCOMPGYRO
```

File: src/drivers/ins/vectornav/VectorNav.cpp [hunk 4]
```diff
@@ -0,0 +1,657 @@
...
+### Examples
+
+Attempt to start driver on a specified serial device.
+$ vectornav start -d /dev/ttyS1
+Stop driver
+$ vectornav stop
+)DESCR_STR");
+
+	PRINT_MODULE_USAGE_NAME("vectornav", "driver");
+	PRINT_MODULE_USAGE_SUBCATEGORY("distance_sensor");
```
Comment (dagar) [inline src/drivers/ins/vectornav/VectorNav.cpp:644]:
```suggestion
	PRINT_MODULE_USAGE_SUBCATEGORY("ins");
```

File: ROMFS/px4fmu_common/init.d-posix/rcS [hunk 5]
```diff
@@ -318,5 +318,9 @@ else
 fi
 . ${R}etc/init.d/rc.logging
 
+
+vectornav start -d /dev/ttyUSB0
```
Comment (dagar) [inline ROMFS/px4fmu_common/init.d-posix/rcS:322]:
```suggestion
```

serial configuration

**Comment by ajordan5** [comment]
I have a VN-100 I'd love to try integrating with my pixhawk. Is this branch in working order? If so, what do I need to do to get the pixhawk to start using my VN100 for estimation?

I'm happy to test and contribute in any way needed!

**Comment by qaz6906** [comment]
After downloading PX4 1.14.0 beta version, I tested it on pixhawk 6X.
Additionally, by referring to the source code of PX4, BinaryOutput and divider settings were completed in the vectornav control center. (Photo attached)
In addition, the following settings were made through the vectornav control center.
baudrate : 921600
ASCII : OFF
VPE: enable
VPE heading mode: Absolute

px4_1
px4_2
px4_3
vn_1
vn_2_1
vn_2_2
vn_2_3
vn_2_4
vn_3_1
vn_3_2

First, we confirmed that Gyro data and Accel data are normally received.

However, the GPS data is derived very strangely, and I am in Korea, but when I check the map of QGC, it says that I am in Africa. (QGC photo and ulog of PX4 attached)
When I checked the test_log_vehicle_gps_position_0.csv file, it seemed like the latitude, longitude, and altitude values were misplaced.

normal situation
latitude = VN GNSS latitude data
longitude = VN GNSS longitude data
Altitude = VN GNSS Altitude Data

Current derived data (unusual situation)
latitude = ?
longitude = VN GNSS latitude data
Altitude = VN GNSS longitude data

[log data.zip](https://github.com/PX4/PX4-Autopilot/files/11133786/log.data.zip)
QGC test_1

It is assumed that there is a bug in the decoding process part of the PX4 source code.

And the PX4 1.14.0 beta version used for the test has a problem with not recognizing the RC receiver properly, so it is regrettable that it is impossible to test through the flight test. (Futaba 7008SB receiver not working on pixhawk 6X)
I would appreciate it if you could reply after checking that part.

**Pull request merged**